File: java/parser/src/main/java/gov/lbl/parser/domain/Der_class_specifier_value.java
package gov.lbl.parser.domain;

import java.util.List;

public class Der_class_specifier_value {
    private Name type_specifier;
    private List<String> identifiers;
    private Comment comment;

    public Der_class_specifier_value(Name type_specifier, List<String> identifiers, Comment comment) {
        this.type_specifier = type_specifier;
        this.identifiers = identifiers;
        this.comment = comment;
    }
}

File: json2mo/derClassSpecifierValue.js
function parse (content, rawJson = false) {
  const util = require('util')
  const nameParser = require('./name')
  const commentParser = require('./comment')

  let moOutput = 'der('
  if (content.type_specifier != null) {
    moOutput += nameParser.parse(content.type_specifier, rawJson)
  }
  let identifiers = null
  if (rawJson) {
    identifiers = content.identifiers
  } else {
    identifiers = content.identifier
  }
  if (identifiers != null) {
    identifiers.forEach(identifier => {
      moOutput += util.format('%s, ', identifier)
    })
    moOutput = moOutput.slice(0, -2)
    moOutput += ') '
  }
  if (rawJson) {
    if (content.comment != null) {
      moOutput += commentParser.parse(content.comment, rawJson)
    }
  } else {
    if (content.description != null) {
      moOutput += commentParser.parse(content.description, rawJson)
    }
  }

  return moOutput
}

module.exports = { parse }
